### ISSUE TITLE
[bug] FTB Quests not accurately counting the number of kills

### ISSUE THREAD

**Issue by ClaudiusMinimus** [FTBTeam/FTB-Quests#604]
To help us help you, please provide as much information as you can, this helps us with debugging and reproducing the issue.

**Mod & Pack version**
ftb-quests-1605.2.1-build.39-forge

**Forge / Fabric version, if applicable**
Forge 36.1.25
**Modpack (with version), if applicable**
[Create on a Potato PC 1.16.5-1.22.0](https://www.curseforge.com/minecraft/modpacks/create-on-a-potato/files/3349473)

We are having an issue with FTB Quests not accurately counting the number of kills for mobs. For example, I have a player whose stats show he has killed 8 Ender Dragons, but quests only say 4. He has also killed 2105 Wither Skeletons and quests shows only 956. Once the quests reached 956 kills it stopped registering new kills. What logs can I provide to show you this issue?

![Dragon Stats](https://cdn.discordapp.com/attachments/434387425851342858/855180329428713492/unknown.png)

![](https://cdn.discordapp.com/attachments/434387425851342858/855180430260174849/unknown.png)

![](https://cdn.discordapp.com/attachments/434387425851342858/855180639635243038/unknown.png)

![](https://cdn.discordapp.com/attachments/434387425851342858/855180497189077033/unknown.png)

**Comment by ClaudiusMinimus** [comment]
Will this issue be addressed? I have players pressing me to get this fixed. 
Thank you, 
CM

**Comment by MaxNeedsSnacks** [comment]
The kill count should only actually start counting once they **unlock** the relevant quest? Is this the case here?

**Comment by aaronhowser1** [comment]
Right, it's not a stat task. The counter only starts once the quest is available.

**Comment by ClaudiusMinimus** [comment]
> The kill count should only actually start counting once they **unlock** the relevant quest? Is this the case here?

It may be the case for when the quest starts, but that is not the issue. The issue is that once the quest starts it seems to stop counting properly kills once the number approaches 1,000.  in the case of the Wither Skeletons. This player had killed prerequisite 10, then 100, and when he started the 1,000 kill quest it showed his number what seemed to be correct. However, as he approached 1,000 kills it seemed not all kills were counted and now he has killed over 2,000 Wither Skeletons and still not completed the quest.

Regarding the dragon kills: I know he has killed at least 20-30 dragons and he tells me the quest is still not complete.

I have included the quest file for you to see.
[hunter.snbt.zip](https://github.com/FTBTeam/FTB-Quests/files/6749440/hunter.snbt.zip)

**Comment by aaronhowser1** [comment]
Did he kill those 2,000 wither skeletons after the quest was unlocked? Or did he kill 956 after it was unlocked? If he kills more, does the number go up?

**Comment by ClaudiusMinimus** [comment]
Thank you for your reply!

Most were killed after the quest was unlocked. If he kills more the number may go up, but it was stuck at 947 for quite a while. At the time of the initial post 14 days ago it was stuck on 956. I should also mention this is not the first player to report this issue and it is not specific to just this modpack.
